### PR TITLE
Pronto implementation

### PR DESCRIPTION
**Pull request by bengtmartensson**
This PR implements `IRrecv::dumpPronto` and `IRsend::sendPronto`. There is a new example that demonstrates `sendPronto` (both for signals in RAM, PROGMEM, and the F()-form). The example `IRrecvDumpV2` has been extended to generate output from `dumpPronto`. 

[Reference](http://www.harctoolbox.org/Glossary.html#ProntoSemantics)

### CODE REVIEW

File: examples/IRrecvDumpV2/IRrecvDumpV2.ino [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #if defined(ESP32)
 int IR_RECEIVE_PIN = 15;
 #else
-int IR_RECEIVE_PIN = 11;
+int IR_RECEIVE_PIN = 5;
```
Comment (ArminJo) [inline examples/IRrecvDumpV2/IRrecvDumpV2.ino:12]: Why this change?
It is inconsistent with all other examples then.

Comment (bengtmartensson) [reply examples/IRrecvDumpV2/IRrecvDumpV2.ino]: Just forgot to set it back. No reason.

Comment (ArminJo) [reply examples/IRrecvDumpV2/IRrecvDumpV2.ino]: Thanks

**Comment by ArminJo** [comment]
Hi Bengt,
Thanks for the code, I cant realy test it, and do not know much about Pronto, but it looks fine so far.
Would you like to sleep one night before merging (my approach) or should I merge the PR now?
Best Regards
Armin

**Comment by bengtmartensson** [comment]
Armin, take your time. Best would be if someone can write a review,  @probonopd ??

**Comment by ArminJo** [comment]
I need no time 😀  I make what you propose!
My experience is, that the developer (mostly me) should wait for 24 hours before releasing a bigger change.
@probonopd It would be fantastic if you can review the code.

**Comment by probonopd** [comment]
When it comes to reviewing C++ I'm the wrong guy, sorry, but I really trust what @bengtmartensson is doing.
